### ISSUE TITLE
There is a problem with right-clicking the tab label

### ISSUE THREAD

**Issue by liuanlin-mx** [eranif/codelite#2688]
Linux Mint 20.1 MATE 64bit 1920x1080
codelite 15
1. tabs > 10
2. ctrl-tab  switch  
3. right-clicking  tab
void clGTKNotebook::GTKRightDown()
{
    long flags = 0;
    wxPoint pt = ScreenToClient(::wxGetMousePosition());
    int where = HitTest(pt, &flags); <<<< Sometimes it's wrong

Another problem：
windows:
![f1](https://user-images.githubusercontent.com/20225987/110666318-2f4d3080-8204-11eb-8c98-dca804e6125c.png)

linux:
![f2](https://user-images.githubusercontent.com/20225987/110666339-35dba800-8204-11eb-853c-f4e97c4d25fe.png)

**Comment by eranif** [comment]
I found a pattern:
if you right click on the tab which can not fit into the screen (i.e. it requires to be scrolled), the last possible tab will get selected

```
[ tab 1][tab 2]..[tab N][overflow tab 1]...[overflow tab N]
```
Now, if you right click any of the `overflow` tabs, the last tab which can be fitted into screen `[tab N]` will get selected for some reason

Still need to figure why this is happening

**Comment by eranif** [comment]
OK, I found the problem. It's seems like a GTK3 bug (or the way wxWidgets is using the API):
When tabs are overflowing the tab area (i.e. the right/left arrows appear in the notebook)
Now, wxWidgets has an internal method which helps determine which tab was clicked, it does that by going over all the tabs and obtaining the rectangle for the tab label at position i, so incase the current mouse is inside that rectangle -> this is our selection.

However, if we play around with the tabs in a way that a tab that was previously visible is now hidden and then we qurey the rectangle for that specific tab label, we will get a valid tab label rectangle!

one would expect that hidden labels (i.e. non visible tabs on screen) - will return an invalid rectangle `(-1,-1,-1,-1)` (this is indeed the case when we first load the notebook), but this is not the case and the source for this bug

I implemented a different logic inside CodeLite's notebook to fix this (basically it does that):

- Find the active tab
- Move to the right as long as the `x` coordinate of the n-th tab is growing, once it does not not stop (i.e. we found the last visible tab on the right side)
- Move to the left as long as the `x` coordinate of the n-th  tab is decreasing, once it does not, stop (i.e. we found the first visible tab)
Now that we have a valid list of visible tabs, we can determine which tab was clicked by checking the mouse position against these tabs only
